src/crimson/os/seastore/scanner.h

```diff
@@ -0,0 +1,112 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:nil -*-
+// vim: ts=8 sw=2 smarttab expandtab
+
+#pragma once
+
+#include "crimson/common/errorator.h"
+#include "crimson/os/seastore/seastore_types.h"
+
+namespace crimson::os::seastore {
+
+class SegmentManager;
+class SegmentCleaner;
+
```

Comment:
I'm ok with leaving this for now, but ultimately, I want all logic for writing, encoding, and scanning a segment to be encapsulated here.  Journal and ExtentPlacementManager use the same format, and I'd like all of the logic for that the be unified.

Reply:
Do you mean that we change the current Scanner class to be a utility class or namespace for `record_t` writing/encoding/scanning/reading in the future?

Reply:
record_t and also the segment header, yes.  Namespace should probably be seastore::segment_encoding or something rather than Scanner, though.  Ideally, no other component would need to know the layout or encoding of a segment.

Reply:
We can complete that refactor in a later PR though.